samples/snippets/csharp/VS_Snippets_Wpf/ContentControlTemplateSelector/CSharp/Window1.xaml.cs

```diff
@@ -32,40 +32,27 @@ public Window1()
     //<Snippet2>
     public class NumderDataTemplateSelector : DataTemplateSelector
     {
+        public DataTemplate NumberTemplate { get; set; }
+        public DataTemplate LargeNumberTemplate { get; set; }
+
         public override DataTemplate SelectTemplate(object item, DependencyObject container)
         {
-            string numberStr = item as string;
-
-            if (numberStr != null)
+            //null value will be used in design mode
+            if (item == null) return null;
+
+            var num = Convert.ToInt32((string)item);
```

Comment:
This can still produce `FormatException`.

Reply:
The try catch should be there.

Reply:
According to example description there is no any values in this example that can produce FormatException and this TemplateSelector does not have meaning in working with other data types and data formats.
If you are talking about future changes to this code, note that in fact this can also produce OutOfMemoryException, InvalidCastException, ThreadAbortException and any type of exception in this lines as well as in any other lines. Catching any of exceptions we do not expect will lead to an issue, silently caught and masked by the behavior implemented inside catch block. For example: catch(FormatException){return null;} will produce the behavior not stated in example description and behavior neither user of application nor user (developer) of this class expect.

Statement from c# Programming Guide:
_Do not catch an exception unless you can handle it and leave the application in a known state. If you catch System.Exception, rethrow it using the throw keyword at the end of the catch block._
More information here: https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/exceptions/

So, allowing this code to throw FormatException actually allows user if this type to be notified about wrong usage (as Exceptions in .NET introduced for).

If we want to make this situation better we can write additional code, like:

try
{
   var num = Convert.ToInt32((string)item);
}
catch(InvalidCastException invalidCastException)
{
   throw new InvalidCastException($"Every item passed to this template must be of type {typeof(String)}",invalidCastException);
}
catch(FormatException formatException)
{
   throw new InvalidCastException($"Every item string passed to this template must be convertible to {typeof(int)}",invalidCastException);
}

This is only a small part of possible exceptional situations, but it shows several times more code than necessary for TemplateSelector explainations.

I see the possibility to create generic TemplateSelector<T> where T must be if type int. In this case it is possible to avoid errors in runtime and to get them in compile time, but again, there will be too much of code.

Reply:
If thats the desired behavior i'm ok with it.

Reply:
Sorry, with what exactly?

Reply:
The exception handling.

Reply:
Anyway, FormatException is not possible in this example, don't you agree?

Reply:
Yes I agree.